crates/aptos-crypto/src/arkworks/shamir.rs

```diff
@@ -3,35 +3,34 @@
 
 //! Contains a version of shamir secret sharing and `ThresholdConfig` for arkworks
 
-use crate::arkworks::{
-    differentiate::DifferentiableFn,
-    serialization::{ark_de, ark_se},
-    vanishing_poly,
+use crate::{
+    arkworks::{differentiate::DifferentiableFn, vanishing_poly},
+    player::Player,
+    traits,
 };
 use anyhow::{anyhow, Result};
+use ark_ec::CurveGroup;
 use ark_ff::{batch_inversion, FftField, Field, PrimeField};
 use ark_poly::{EvaluationDomain, Radix2EvaluationDomain};
 use ark_std::fmt;
+use rand::{seq::IteratorRandom, Rng};
+use rand_core::{CryptoRng, RngCore};
 use serde::{Deserialize, Deserializer, Serialize};
 use std::collections::{HashMap, HashSet};
 
-/// Represents a single share in Shamir's Secret Sharing scheme. Each
-/// `ShamirShare` consists of an `(x, y)` point on the secret sharing polynomial.
-#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq)]
-pub struct ShamirShare<F: PrimeField> {
-    /// The interpolation point of the secret sharing polynomial.
-    #[serde(serialize_with = "ark_se", deserialize_with = "ark_de")]
-    pub x: F,
-    /// The evaluation of the polynomial at `x`.
-    #[serde(serialize_with = "ark_se", deserialize_with = "ark_de")]
-    pub y: F,
-}
-
-/// Configuration for a threshold cryptography scheme. We're restricting F to `Primefield`
-/// because Shamir shares are usually defined over such a field. For reconstructing to a group (TODO)
-/// we'll use a generic parameter `G: CurveGroup<ScalarField = F>`
+/// A pair corresponding to a player (i.e., an index into the roots of unity) and a field element
+/// evaluation
+#[allow(type_alias_bounds)]
+pub type ShamirShare<F: PrimeField> = (Player, F);
+/// A pair corresponding to a player (i.e., an index into the roots of unity) and a group element
+/// whose exponent is an evaluation
+#[allow(type_alias_bounds)]
+pub type ShamirGroupShare<G: CurveGroup> = (Player, G::Affine);
+
+/// Configuration for a threshold cryptography scheme. Usually one restricts `F` to `Primefield`
+/// but any field is theoretically possible.
 #[derive(Debug, Clone, Copy, Serialize, PartialEq, Eq)]
-pub struct ThresholdConfig<F: PrimeField> {
+pub struct ShamirSharingScheme<F: FftField> {
```

Comment:
I prefer this be called `ShamirThresholdConfig` for extra accuracy (i.e., threshold) and consistency  (i.e., config).

But I'll keep reading to see if this makes sense.